### PR TITLE
[mir_demo_client_wayland] exit on SIGTERM (and therefore pass smoke test)

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: examples/wayland_client.c [hunk 1]
```diff
@@ -345,6 +346,16 @@ static struct wl_output_listener const output_listener = {
     .scale = &output_scale,
 };
 
+static volatile sig_atomic_t running = 0;
```
Comment (gerboland) [inline examples/wayland_client.c]: When is this ever set to true?

Comment (AlanGriffiths) [reply examples/wayland_client.c]: Fixed

File: examples/wayland_client.c [hunk 2]
```diff
@@ -384,7 +395,12 @@ int main()
 
     wl_output_add_listener(globals->output, &output_listener, NULL);
 
-    while (wl_display_dispatch(display))
+    signal(SIGINT, shutdown);
+    signal(SIGTERM, shutdown);
+    signal(SIGHUP, shutdown);
+    running = 1;
```
Comment (gerboland) [inline examples/wayland_client.c]: Nitpick: I expect it's possible for the signal handler to be fired before here, so it'll print Goodbye but still enter the while loop. Perhaps better to set running to true on initiation?

Comment (AlanGriffiths) [reply examples/wayland_client.c]: It only prints goodbye when running is set

Comment (gerboland) [reply examples/wayland_client.c]: Yes you're right. Ignore me

**Pull request merged**